bluemira/magnetostatics/semianalytic_2d.py

```diff
@@ -113,6 +123,18 @@ def _full_z_integrand(phi, r1, r2, z1, z2):
     )
 
 
+def _integrate_z_by_parts(r1, r2, z1, z2):
+    """
+    Fall back to integrating the Bz integrand by parts if integration fails.
```

Comment:
Private function, so it doesn't really matter, but this docstring feels slightly odd. By reading just the docstring, I would assume the function tries to integrate, then falls back to integrating by parts.

Should it read something like the below?

```python
"""
Integrate the Bz integrand by parts.

This can be used as a fall back if normal integration fails.
"""